src/platform/linux.rs

```diff
@@ -824,56 +862,107 @@ mod desktop {
+                log::debug!("extracting xauth file for Xorg running with uid: {}", self.uid);
                 for line in output.lines() {
                     let mut auth_found = false;
-                    for v in line.split_whitespace() {
-                        if v == "-auth" {
-                            auth_found = true;
-                        } else if auth_found {
-                            if std::path::Path::new(v).is_absolute() {
-                                self.xauth = v.to_string();
-                            } else {
-                                if let Some(pid) = line.split_whitespace().nth(1) {
-                                    let home_dir = get_env_from_pid("HOME", pid);
-                                    if home_dir.is_empty() {
-                                        self.xauth = format!("/home/{}/{}", self.username, v);
+
+                    if let Some(pid) = line.split_whitespace().nth(1) {
+                        for v in line.split_whitespace() {
+                            if v == "-auth" {
+                                log::debug!("- auth parameter detected");
+                                auth_found = true;
+                            } else if auth_found {
+                                if std::path::Path::new(v).is_absolute()
+                                    && std::path::Path::new(v).exists() {
+
+                                    self.xauth = v.to_string();
+                                    log::debug!("+ auth file with absolute path detected: {}", self.xauth);
+                                } else {
+                                    log::debug!("- auth file with relative path detected: {}", v);
+                                    if let Some(pid) = line.split_whitespace().nth(1) {
+                                        let mut base_dir: String = String::from("/home"); // default pattern
+                                        let home_dir = get_env_from_pid("HOME", pid);
+                                        if home_dir.is_empty() {
+                                            if let Some(home) = get_user_home_by_name(&self.username) {
+                                                base_dir = home.as_path().to_string_lossy().to_string();
+                                            };
+                                        } else {
+                                            base_dir = home_dir;
+                                        }
+                                        if Path::new(&base_dir).exists() {
+                                            self.xauth = format!("{}/{}", base_dir, v);
+                                            log::debug!("+ auth file with home path added: {}", self.xauth);
+                                        };
                                     } else {
-                                        self.xauth = format!("{}/{}", home_dir, v);
+                                        // unreachable!
```

Comment:
Can you explain what changes here line870 -> line 895?

Reply:
Highlevel: 
As mentioned by newly added debug messages auth- file will be extraxcted from Xorg process and for a relative path home path will be added.

If no HOME var was found the home folder for given user will be determined from system config (new function: get_user_home_by_name)  old logic just used hard coded  default pattern "/home/<username>" (now only if no system config was found)  which wasn't working for domain users (AD, samba, freeipa, ...) where an additional domain subfolder (/home/<domain>/<username>) needs to be added.

In addition detected file needs to exist, if not this loop won't detect any auth- file. So next step can detect a valid file.

Next step (Todo): Detect auth file by checking process tree (processes below Xorg) without need of predefined checks for Gname, XFCE, Plasma, ...). But as this was my first modifiaction using rust language, there was too less time for better solution.